client/state/data-layer/wpcom/activity-log/update-credentials/index.js

```diff
@@ -70,7 +72,11 @@ export const request = ( action ) => {
 				apiNamespace: 'wpcom/v2',
 				method: 'POST',
 				path: `/sites/${ action.siteId }/rewind/credentials/update`,
-				body: { credentials },
+				body: { credentials, stream: true },
+				expectStreamMode: true,
+
+				// TODO @azabani make this a requestDispatcher option to fully integrate with wpcom-http
+				onStreamRecord: ( record ) => debug( 'onStreamRecord: record=%o', record ),
```

Comment:
(note that i’ve done exactly that in #47614)

Reply:
Yes, if you want to be compatible with the data layer philosophy, then you don't use `onStreamRecord` as a callback. It should be a Redux action.

And you'll also need to enhance `dispatchRequest` to include the `onStreamRecord` option.

You want to look at how `onProgress` events flow through the code and implement something very similar for `onStreamRecord`.

The `wpcom.req` handler gets a `onStreamRecord` callback that dispatches an extended Redux action:
```js
const onStreamRecord = ( record ) => {
  dispatch( extendAction( action.onStreamRecord, streamRecordMeta( record ) ) );
}
```

That dispatches the `action.onStreamRecord` action (it's an object with `type` etc.), enhanced with a `meta.dataLayer.streamRecord` property.

The `dispatchRequest` handler then intercepts this action, and needs to detect that there is the `meta.dataLayer.streamRecord` meta (in `createRequestAction`, with a check similar to `getProgress`). Then it will call its `onStreamRecord` action creator callback that receives `action` and `record` as arguments, and which returns an app-level action to dispatch and handle the record.

There are three functions/objects called `onStreamRecord`, and each of them is a different thing 🙁

Reply:
thanks for the detailed walkthrough! i think #47614 should be aligned with every step of this, but let me know if i’ve missed anything.